sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/FileUtilController.java

```diff
@@ -66,7 +71,28 @@ public FileUtilController(
     @PostMapping("api/upload")
     @Override
     public String uploadFile(@RequestHeader("FileType") Type type, @RequestParam("file") MultipartFile file) throws IOException {
-        return uploadService.upload(type, file.getInputStream(), file.getOriginalFilename(), file.getSize());
+        String resourceURL;
+
+        // Performing size (less than 2 MB) and file type (JPG/PNG only) validation for images
+        if (COMMON.equals(type) || USERS.equals(type)) {
+
+            boolean isSizeLessThan2Mb = file.getSize() <= 2_000_000;
+            if (!isSizeLessThan2Mb) {
+                throw new IllegalOperationException(INVALID_FILE_SIZE, "File exceeds permitted size");
+            }
+
+            String contentType = file.getContentType();
+            boolean isFormatImage =  contentType!= null &&
+                    (contentType.equals("image/jpeg") || contentType.equals("image/png"));
+            if (!isFormatImage) {
+                throw new IllegalOperationException(INVALID_FILE_FORMAT, "File format " + contentType + " is not permitted, try JPEG or PNG");
+            }
+
+            resourceURL = uploadService.uploadImages(type, file.getInputStream(), file.getOriginalFilename(), file.getSize());
+        } else {
+            resourceURL = uploadService.uploadArtifacts(type, file.getInputStream(), file.getOriginalFilename(), file.getSize());
+        }
+        return resourceURL;
```

Comment:
Finally: combine 2 checks into one. Pseudocode:
```java
if (file.getSize() >= 2_000_000 || ALLOWED_CONTENT_TYPES.contains(file.getContentType())) {
   throw new ... // generic exception message on both size and format constraints.
}
```
Your controller will become elegant and beautiful in no time :)